src/addDefaultParameters.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Codacyrc,
+  Parameter,
+  ParameterSpec,
+  Pattern,
+  PatternSpec,
+  Specification
+} from "."
+
+export function addDefaultParameters(
+  codacyrc: Codacyrc,
+  specification?: Specification
+) {
+  const tools = codacyrc.tools ?? []
+  const specificationPatterns = specification ? specification.patterns : []
+
+  tools.map((tool) => {
+    const patterns = tool.patterns ?? []
+    patterns.map((pattern) =>
+      addDefaultParamentersToPattern(pattern, specificationPatterns)
+    )
+  })
+}
```

Comment:
You're using map without return anything, it should be `forEach` if you maintain this semantics.
What I suggest you, however, is to return a new object with something like this:
```suggestion
  return tools.map((tool) => {
    const patterns = tool.patterns ?? []
    return patterns.map((pattern) =>
      return withDefaultParameters(pattern, specificationPatterns)
    )
  })
}
```

Reply:
But that wouldn't work right? You would need to make some "copy" like tool.copy( or something?

Reply:
For the arrays it should work, for the objects you need to copy them as you said. There are some examples either in this repository or in codacy-eslint about it.

Reply:
The only example that I could find was for mutable data, the "copy" will still return immutable data.. You will help me Monday on this right? 👍